### PR TITLE
Fix/custom provider

### LINKED ISSUE #14: Replace injected web3 use with Provider.
- Will allow Drizzle to set the `web3` version by instantiating its own.
- Will stop race condition bug where `defaultAccount` is not set in time for a transaction without a `from` field, causing that transaction to fail with confusing error messages; i.e. "incorrect nonce".

### PR DESCRIPTION
**Pull request by cds-amal**
This PR fixes #14. 

# How to test

**Note you will need 2 terminals**

## First terminal: 
- build out monorepo from the root project and start the ganache server
`   yarn && yarn ganache`

## Second terminal:
- deploy test contracts and start the test app
```
  yarn ganache:deploy
  cd ui-tests/custom-provider
  yarn serve:ui
```

## Verify 
Use the browser of your choice with metamask installed, visit [the test app](htpt://localhost:3003) and change SimpleStorage's StoredValue. The test is successful if you're able to change the value without MetaMask showing up.

### CODE REVIEW

**Comment by OliveIT** [comment]
Hi @cds-amal 
Webpack shows this warning when I try to build. ;)
```
WARNING in ./src/web3/web3Saga.js 85:18-35
"export 'WEB3_ERROR' (imported as 'Action') was not found in './constants'
 @ ./src/drizzleStatus/drizzleStatusSaga.js
 @ ./src/index.js
```

**Comment by cds-amal** [comment]
Thanks @OliveIT, thanks for testing and providing feedback! That issue is in another branch and will make it in as well.

**Comment by cds-amal** [comment]
@DiscRiskandBisque pointed out that customProvider should be an Object (a valid provider), not a url.

**Comment by adrianmcli** [comment]
Test works great, gonna take a look at the code and if nothing stands out let's merge!

File: packages/store/test/web3.test.js [hunk 1]
```diff
@@ -17,30 +17,17 @@ describe('Resolving Web3', () => {
...
 
       // First action dispatched
       expect(gen.next().value).toEqual(put({ type: Action.WEB3_INITIALIZED }))
 
       resolvedWeb3 = gen.next().value
-
-      hasWeb3Shape(resolvedWeb3)
-    })
-
-    test('get network ID', async () => {
```
Comment (adrianmcli) [inline packages/store/test/web3.test.js:34]: @cds-amal just wondering why this test was removed.

**Comment by adrianmcli** [comment]
@cds-amal just the one question above, otherwise feel free to merge.

File: packages/store/test/web3.test.js (hunk 1 above)
Comment (cds-amal) [reply packages/store/test/web3.test.js:34]: Checking the network id is an arbitrary test for a custom provider.

**Pull request merged**